File: Distancia2puntosC3d/Distancia2puntosC/PuntoXYZ.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Distancia2puntosC
{
    class PuntoXYZ:PuntoXY 
    {
        protected double z;
        public double Z
        {
            get { return z; }
            set { z = value; }
        }
        public PuntoXYZ() : base()
        {
            z = 0;
        }
        public PuntoXYZ(double X, double Y,double Z) : base(X,Y)
        {
            this.z = Z;
        }

        public double CalcularDistancia(PuntoXYZ a)
        {
            return Math.Sqrt(Math.Pow((x - a.X), 2) + Math.Pow((y - a.Y), 2) + Math.Pow((z - a.Z), 2));
        }
        

        public override string ToString()
        {
            return x + " ," + y +" , "+z;
        }
    }
}

File: Distancia2PuntosVB3d/Distancia2PuntosVB/PuntoXYZ.vb
Public Class PuntoXYZ
    Inherits PuntoXY

    Protected z As Double

    Public Property _Z As Double
        Get
            Return z
        End Get
        Set(ByVal value As Double)
            z = value
        End Set
    End Property

    Public Sub New()
        MyBase.New()
        z = 0
    End Sub

    Public Sub New(ByVal _X As Double, ByVal _Y As Double, ByVal _Z As Double)
        MyBase.New(_X, _Y)
        Me.z = _Z
    End Sub

    Public Function CalcularDistancia(ByVal a As PuntoXYZ) As Double
        Return Math.Sqrt(Math.Pow((x - a._X), 2) + Math.Pow((y - a._Y), 2) + Math.Pow((z - a._Z), 2))
    End Function

    Public Overrides Function ToString() As String
        Return x & " ," + y & " , " + z
    End Function
End Class
